Sources/ImageKit/ImageKit.swift

```diff
@@ -24,9 +23,10 @@ open class ImageKit: NSObject {
     var configured = false
     lazy var userDefaults = UserDefaults.standard
 
-    public static let shared = ImageKit()
+    nonisolated(unsafe) public static let shared = ImageKit()
     
-    private let sharedUploader = ImageKitUploader()
+    @MainActor
+    private lazy var sharedUploader = ImageKitUploader()
```

Comment:
Are we sure we should run the uploader on the main thread?

AFAIK, MainActor should be used on UI-bound work. Network I/O and any heavy pre-processing (resizing, JPEG/HEIC encoding) should run off the main actor to avoid jank.

Reply:
Understood